src/lib.rs

```diff
@@ -198,119 +198,148 @@ mod sym;
 mod debug;
 
 use proc_macro::TokenStream;
+use std::sync::atomic::{AtomicUsize, Ordering::SeqCst};
 
-/// A #\[proc_macro\] implemented in wasm!
-///
-/// # Canonical macro implementation:
-///
-/// ```
-/// # const IGNORE: &str = stringify! {
-/// use proc_macro2::TokenStream;
-///
-/// #[no_mangle]
-/// pub extern "C" fn my_macro(input: TokenStream) -> TokenStream {
-///     proc_macro2::set_wasm_panic_hook();
-///
-///     ...
-/// }
-/// # };
-/// ```
-///
-/// # Canonical entry point:
-///
-/// ```
-/// # const IGNORE: &str = stringify! {
-/// extern crate proc_macro;
-///
-/// use proc_macro::TokenStream;
-///
-/// static WASM: &[u8] = include_bytes!("my_macro.wasm");
-///
-/// #[proc_macro]
-/// pub fn my_macro(input: TokenStream) -> TokenStream {
-///     watt::proc_macro("my_macro", input, WASM)
-/// }
-/// # };
-/// ```
-pub fn proc_macro(fun: &str, input: TokenStream, wasm: &[u8]) -> TokenStream {
-    exec::proc_macro(fun, vec![input], wasm)
+pub struct Instance {
```

Comment:
It'd probably be good to have a doc comment on the `Instance` struct itself, explaining that it's intended to be used in a static.

Reply:
👍